### ISSUE TITLE
Expire not following time based retention period with multiple repos

### ISSUE THREAD

**Issue by Tostino** [pgbackrest/pgbackrest#1852]
Please provide the following information when submitting an issue (feature requests or general comments can skip this): Expire not following time based retention period with multiple repos

1. pgBackRest version: 2.39

2. PostgreSQL version: PostgreSQL 10.21 (Debian 10.21-1.pgdg100+1) on x86_64-pc-linux-gnu, compiled by gcc (Debian 8.3.0-6) 8.3.0, 64-bit

3. Operating system/version - if you have more than one server (for example, a database server, a repository host server, one or more standbys), please specify each: Debian

4. Did you install pgBackRest from source or from a package? Package

5. Please attach the following as applicable:
    - `pgbackrest.conf` file(s) [pgbackrest.conf.txt](https://github.com/pgbackrest/pgbackrest/files/9447440/pgbackrest.conf.txt)
    - `postgresql.conf` settings applicable to pgBackRest (`archive_command`, `archive_mode`, `listen_addresses`, `max_wal_senders`, `wal_level`, `port`)
    - errors in the postgresql log file before or during the time you experienced the issue
    - log file in `/var/log/pgbackrest` for the commands run (e.g. `/var/log/pgbackrest/mystanza_backup.log`)

7. Describe the issue: I am trying to determine what my issue is here, we have multiple repos (posix / s3) which have different retention periods, and something isn't interacting as intended.
I wanted my s3 repo (repo2) to retain full backups for 365 days while retaining 1 most recent diff, and my posix repo (repo1) to retain full backups for 14 days while keeping my 3 most recent diffs.

Instead it is expiring my s3 (repo2) backups way before they should be.

Logs for expired backups:
```
-------------------PROCESS START-------------------
2022-08-14 04:02:12.076 P00   INFO: expire command begin 2.39: --exec-id=2532-01d14fab --log-level-file=detail --repo1-cipher-pass=<redacted> --repo2-cipher-pass=<redacted> --repo1-cipher-type=aes-256-cbc --repo2-cipher-type=aes-256-cbc --repo1-path=/mnt/backup/pgbackrest --repo2-path=/dev/backup/oltp/pgbackrest --repo1-retention-diff=3 --repo2-retention-diff=3 --repo1-retention-full=14 --repo2-retention-full=365 --repo1-retention-full-type=time --repo2-retention-full-type=time --repo2-s3-bucket=tpm-backup --repo2-s3-endpoint=nyc3.digitaloceanspaces.com --repo2-s3-key=<redacted> --repo2-s3-key-secret=<redacted> --repo2-s3-region=nyc3 --repo1-type=posix --repo2-type=s3 --stanza=dev-oltp
2022-08-14 04:02:12.079 P00   INFO: repo1: expire diff backup 20220801-040009F_20220811-040005D
2022-08-14 04:02:12.200 P00   INFO: repo1: remove expired backup 20220801-040009F_20220811-040005D
2022-08-14 04:02:12.236 P00   INFO: repo1: time-based archive retention not met - archive logs will not be expired
2022-08-14 04:02:13.220 P00   INFO: repo2: time-based archive retention not met - archive logs will not be expired
2022-08-14 04:02:13.320 P00 DETAIL: statistics: {"http.client":{"total":2},"http.request":{"total":11},"http.session":{"total":2},"socket.client":{"total":2},"socket.session":{"total":2},"tls.client":{"total":2},"tls.session":{"total":2}}
2022-08-14 04:02:13.320 P00   INFO: expire command end: completed successfully (1245ms)
-------------------PROCESS START-------------------
2022-08-15 04:02:41.453 P00   INFO: expire command begin 2.39: --exec-id=19236-fc081f57 --log-level-file=detail --repo1-cipher-pass=<redacted> --repo2-cipher-pass=<redacted> --repo1-cipher-type=aes-256-cbc --repo2-cipher-type=aes-256-cbc --repo1-path=/mnt/backup/pgbackrest --repo2-path=/dev/backup/oltp/pgbackrest --repo1-retention-diff=3 --repo2-retention-diff=3 --repo1-retention-full=14 --repo2-retention-full=365 --repo1-retention-full-type=time --repo2-retention-full-type=time --repo2-s3-bucket=tpm-backup --repo2-s3-endpoint=nyc3.digitaloceanspaces.com --repo2-s3-key=<redacted> --repo2-s3-key-secret=<redacted> --repo2-s3-region=nyc3 --repo1-type=posix --repo2-type=s3 --stanza=dev-oltp
2022-08-15 04:02:41.458 P00   INFO: repo1: expire diff backup 20220801-040009F_20220812-040007D
2022-08-15 04:02:41.660 P00   INFO: repo1: remove expired backup 20220801-040009F_20220812-040007D
2022-08-15 04:02:41.763 P00   INFO: repo1: time-based archive retention not met - archive logs will not be expired
2022-08-15 04:02:42.938 P00   INFO: repo2: time-based archive retention not met - archive logs will not be expired
2022-08-15 04:02:43.039 P00 DETAIL: statistics: {"http.client":{"total":2},"http.request":{"total":11},"http.session":{"total":2},"socket.client":{"total":2},"socket.session":{"total":2},"tls.client":{"total":2},"tls.session":{"total":2}}
2022-08-15 04:02:43.039 P00   INFO: expire command end: completed successfully (1586ms)
-------------------PROCESS START-------------------
2022-08-16 04:02:10.088 P00   INFO: expire command begin 2.39: --exec-id=4167-2f8ce999 --log-level-file=detail --repo1-cipher-pass=<redacted> --repo2-cipher-pass=<redacted> --repo1-cipher-type=aes-256-cbc --repo2-cipher-type=aes-256-cbc --repo1-path=/mnt/backup/pgbackrest --repo2-path=/dev/backup/oltp/pgbackrest --repo1-retention-diff=3 --repo2-retention-diff=3 --repo1-retention-full=14 --repo2-retention-full=365 --repo1-retention-full-type=time --repo2-retention-full-type=time --repo2-s3-bucket=tpm-backup --repo2-s3-endpoint=nyc3.digitaloceanspaces.com --repo2-s3-key=<redacted> --repo2-s3-key-secret=<redacted> --repo2-s3-region=nyc3 --repo1-type=posix --repo2-type=s3 --stanza=dev-oltp
2022-08-16 04:02:10.098 P00   INFO: repo1: expire time-based backup 20220621-172226F
2022-08-16 04:02:10.104 P00   INFO: repo1: expire diff backup 20220801-040009F_20220813-040004D
2022-08-16 04:02:10.315 P00   INFO: repo1: remove expired backup 20220801-040009F_20220813-040004D
2022-08-16 04:02:10.381 P00   INFO: repo1: remove expired backup 20220621-172226F
2022-08-16 04:04:26.621 P00 DETAIL: repo1: 10-1 archive retention on backup 20220801-040009F, start = 0000000600005D3E00000055
2022-08-16 04:04:26.731 P00   INFO: repo1: 10-1 remove archive, start = 0000000600005D3D, stop = 0000000600005D3E00000054
2022-08-16 04:04:27.293 P00   INFO: repo2: expire time-based backup 20220714-172548F
2022-08-16 04:04:27.798 P00   INFO: repo2: remove expired backup 20220714-172548F
2022-08-16 04:35:43.799 P00 DETAIL: repo2: 10-1 archive retention on backup 20220801-125242F, start = 0000000600005D3E00000057
2022-08-16 04:35:47.221 P00   INFO: repo2: 10-1 remove archive, start = 0000000600005D3D, stop = 0000000600005D3E00000056
2022-08-16 04:35:47.308 P00   WARN: remote-0 process on 'oltp02' terminated unexpectedly [42]: ERROR: [042]: timeout after 1830000ms waiting for read from 'remote-0'
2022-08-16 04:35:47.308 P00 DETAIL: statistics: {"http.client":{"total":2},"http.request":{"total":399},"http.retry":{"total":1},"http.session":{"total":4},"socket.client":{"total":2},"socket.session":{"total":4},"tls.client":{"total":2},"tls.session":{"total":4}}
2022-08-16 04:35:47.308 P00   INFO: expire command end: completed successfully (2017220ms)
-------------------PROCESS START-------------------
2022-08-17 04:01:58.093 P00   INFO: expire command begin 2.39: --exec-id=21969-fdd4e453 --log-level-file=detail --repo1-cipher-pass=<redacted> --repo2-cipher-pass=<redacted> --repo1-cipher-type=aes-256-cbc --repo2-cipher-type=aes-256-cbc --repo1-path=/mnt/backup/pgbackrest --repo2-path=/dev/backup/oltp/pgbackrest --repo1-retention-diff=3 --repo2-retention-diff=3 --repo1-retention-full=14 --repo2-retention-full=365 --repo1-retention-full-type=time --repo2-retention-full-type=time --repo2-s3-bucket=tpm-backup --repo2-s3-endpoint=nyc3.digitaloceanspaces.com --repo2-s3-key=<redacted> --repo2-s3-key-secret=<redacted> --repo2-s3-region=nyc3 --repo1-type=posix --repo2-type=s3 --stanza=dev-oltp
2022-08-17 04:01:58.100 P00   INFO: repo1: expire diff backup 20220801-040009F_20220814-040003D
2022-08-17 04:01:58.136 P00   INFO: repo1: remove expired backup 20220801-040009F_20220814-040003D
2022-08-17 04:01:58.164 P00   INFO: repo1: time-based archive retention not met - archive logs will not be expired
2022-08-17 04:01:58.791 P00   INFO: repo2: time-based archive retention not met - archive logs will not be expired
2022-08-17 04:01:58.891 P00 DETAIL: statistics: {"http.client":{"total":2},"http.request":{"total":10},"http.session":{"total":2},"socket.client":{"total":2},"socket.session":{"total":2},"tls.client":{"total":2},"tls.session":{"total":2}}
2022-08-17 04:01:58.891 P00   INFO: expire command end: completed successfully (798ms)
-------------------PROCESS START-------------------
2022-08-18 04:01:57.732 P00   INFO: expire command begin 2.39: --exec-id=7579-292dd705 --log-level-file=detail --repo1-cipher-pass=<redacted> --repo2-cipher-pass=<redacted> --repo1-cipher-type=aes-256-cbc --repo2-cipher-type=aes-256-cbc --repo1-path=/mnt/backup/pgbackrest --repo2-path=/dev/backup/oltp/pgbackrest --repo1-retention-diff=3 --repo2-retention-diff=3 --repo1-retention-full=14 --repo2-retention-full=365 --repo1-retention-full-type=time --repo2-retention-full-type=time --repo2-s3-bucket=tpm-backup --repo2-s3-endpoint=nyc3.digitaloceanspaces.com --repo2-s3-key=<redacted> --repo2-s3-key-secret=<redacted> --repo2-s3-region=nyc3 --repo1-type=posix --repo2-type=s3 --stanza=dev-oltp
2022-08-18 04:01:57.739 P00   INFO: repo1: expire diff backup 20220801-040009F_20220815-040010D
2022-08-18 04:01:57.954 P00   INFO: repo1: remove expired backup 20220801-040009F_20220815-040010D
2022-08-18 04:01:58.004 P00   INFO: repo1: time-based archive retention not met - archive logs will not be expired
2022-08-18 04:01:58.733 P00   INFO: repo2: time-based archive retention not met - archive logs will not be expired
2022-08-18 04:01:58.833 P00 DETAIL: statistics: {"http.client":{"total":2},"http.request":{"total":10},"http.session":{"total":2},"socket.client":{"total":2},"socket.session":{"total":2},"tls.client":{"total":2},"tls.session":{"total":2}}
2022-08-18 04:01:58.833 P00   INFO: expire command end: completed successfully (1101ms)
-------------------PROCESS START-------------------
2022-08-19 04:02:09.417 P00   INFO: expire command begin 2.39: --exec-id=26304-c8234332 --log-level-file=detail --repo1-cipher-pass=<redacted> --repo2-cipher-pass=<redacted> --repo1-cipher-type=aes-256-cbc --repo2-cipher-type=aes-256-cbc --repo1-path=/mnt/backup/pgbackrest --repo2-path=/dev/backup/oltp/pgbackrest --repo1-retention-diff=3 --repo2-retention-diff=3 --repo1-retention-full=14 --repo2-retention-full=365 --repo1-retention-full-type=time --repo2-retention-full-type=time --repo2-s3-bucket=tpm-backup --repo2-s3-endpoint=nyc3.digitaloceanspaces.com --repo2-s3-key=<redacted> --repo2-s3-key-secret=<redacted> --repo2-s3-region=nyc3 --repo1-type=posix --repo2-type=s3 --stanza=dev-oltp
2022-08-19 04:02:09.431 P00   INFO: repo1: expire diff backup 20220801-040009F_20220816-040009D
2022-08-19 04:02:09.492 P00   INFO: repo1: remove expired backup 20220801-040009F_20220816-040009D
2022-08-19 04:02:09.551 P00   INFO: repo1: time-based archive retention not met - archive logs will not be expired
2022-08-19 04:02:10.277 P00   INFO: repo2: time-based archive retention not met - archive logs will not be expired
2022-08-19 04:02:10.377 P00 DETAIL: statistics: {"http.client":{"total":2},"http.request":{"total":10},"http.session":{"total":2},"socket.client":{"total":2},"socket.session":{"total":2},"tls.client":{"total":2},"tls.session":{"total":2}}
2022-08-19 04:02:10.377 P00   INFO: expire command end: completed successfully (960ms)
-------------------PROCESS START-------------------
2022-08-19 06:01:53.569 P00   INFO: expire command begin 2.39: --exec-id=9005-082f46ff --log-level-file=detail --repo=2 --repo1-cipher-pass=<redacted> --repo2-cipher-pass=<redacted> --repo1-cipher-type=aes-256-cbc --repo2-cipher-type=aes-256-cbc --repo1-path=/mnt/backup/pgbackrest --repo2-path=/dev/backup/oltp/pgbackrest --repo1-retention-diff=3 --repo2-retention-diff=3 --repo1-retention-full=14 --repo2-retention-full=365 --repo1-retention-full-type=time --repo2-retention-full-type=time --repo2-s3-bucket=tpm-backup --repo2-s3-endpoint=nyc3.digitaloceanspaces.com --repo2-s3-key=<redacted> --repo2-s3-key-secret=<redacted> --repo2-s3-region=nyc3 --repo1-type=posix --repo2-type=s3 --stanza=dev-oltp
2022-08-19 06:01:54.442 P00   INFO: repo2: time-based archive retention not met - archive logs will not be expired
2022-08-19 06:01:54.542 P00 DETAIL: statistics: {"http.client":{"total":2},"http.request":{"total":137},"http.retry":{"total":1},"http.session":{"total":3},"socket.client":{"total":2},"socket.session":{"total":3},"tls.client":{"total":2},"tls.session":{"total":3}}
2022-08-19 06:01:54.542 P00   INFO: expire command end: completed successfully (973ms)
-------------------PROCESS START-------------------
2022-08-20 04:02:00.534 P00   INFO: expire command begin 2.39: --exec-id=11831-f367d715 --log-level-file=detail --repo1-cipher-pass=<redacted> --repo2-cipher-pass=<redacted> --repo1-cipher-type=aes-256-cbc --repo2-cipher-type=aes-256-cbc --repo1-path=/mnt/backup/pgbackrest --repo2-path=/dev/backup/oltp/pgbackrest --repo1-retention-diff=3 --repo2-retention-diff=3 --repo1-retention-full=14 --repo2-retention-full=365 --repo1-retention-full-type=time --repo2-retention-full-type=time --repo2-s3-bucket=tpm-backup --repo2-s3-endpoint=nyc3.digitaloceanspaces.com --repo2-s3-key=<redacted> --repo2-s3-key-secret=<redacted> --repo2-s3-region=nyc3 --repo1-type=posix --repo2-type=s3 --stanza=dev-oltp
2022-08-20 04:02:00.541 P00   INFO: repo1: expire diff backup 20220801-040009F_20220817-040005D
2022-08-20 04:02:00.569 P00   INFO: repo1: remove expired backup 20220801-040009F_20220817-040005D
2022-08-20 04:02:00.607 P00   INFO: repo1: time-based archive retention not met - archive logs will not be expired
2022-08-20 04:02:01.441 P00   INFO: repo2: time-based archive retention not met - archive logs will not be expired
2022-08-20 04:02:01.541 P00 DETAIL: statistics: {"http.client":{"total":2},"http.request":{"total":11},"http.session":{"total":2},"socket.client":{"total":2},"socket.session":{"total":2},"tls.client":{"total":2},"tls.session":{"total":2}}
2022-08-20 04:02:01.541 P00   INFO: expire command end: completed successfully (1007ms)
-------------------PROCESS START-------------------
2022-08-21 04:02:07.360 P00   INFO: expire command begin 2.39: --exec-id=29794-da9dabd6 --log-level-file=detail --repo1-cipher-pass=<redacted> --repo2-cipher-pass=<redacted> --repo1-cipher-type=aes-256-cbc --repo2-cipher-type=aes-256-cbc --repo1-path=/mnt/backup/pgbackrest --repo2-path=/dev/backup/oltp/pgbackrest --repo1-retention-diff=3 --repo2-retention-diff=3 --repo1-retention-full=14 --repo2-retention-full=365 --repo1-retention-full-type=time --repo2-retention-full-type=time --repo2-s3-bucket=tpm-backup --repo2-s3-endpoint=nyc3.digitaloceanspaces.com --repo2-s3-key=<redacted> --repo2-s3-key-secret=<redacted> --repo2-s3-region=nyc3 --repo1-type=posix --repo2-type=s3 --stanza=dev-oltp
2022-08-21 04:02:07.367 P00   INFO: repo1: expire diff backup 20220801-040009F_20220818-040010D
2022-08-21 04:02:07.392 P00   INFO: repo1: remove expired backup 20220801-040009F_20220818-040010D
2022-08-21 04:02:07.448 P00   INFO: repo1: time-based archive retention not met - archive logs will not be expired
2022-08-21 04:02:08.282 P00   INFO: repo2: time-based archive retention not met - archive logs will not be expired
2022-08-21 04:02:08.383 P00 DETAIL: statistics: {"http.client":{"total":2},"http.request":{"total":11},"http.session":{"total":2},"socket.client":{"total":2},"socket.session":{"total":2},"tls.client":{"total":2},"tls.session":{"total":2}}
2022-08-21 04:02:08.383 P00   INFO: expire command end: completed successfully (1023ms)-------------------PROCESS START-------------------
2022-08-22 04:03:30.587 P00   INFO: expire command begin 2.39: --exec-id=15184-b745990c --log-level-file=detail --repo1-cipher-pass=<redacted> --repo2-cipher-pass=<redacted> --repo1-cipher-type=aes-256-cbc --repo2-cipher-type=aes-256-cbc --repo1-path=/mnt/backup/pgbackrest --repo2-path=/dev/backup/oltp/pgbackrest --repo1-retention-diff=3 --repo2-retention-diff=3 --repo1-retention-full=14 --repo2-retention-full=365 --repo1-retention-full-type=time --repo2-retention-full-type=time --repo2-s3-bucket=tpm-backup --repo2-s3-endpoint=nyc3.digitaloceanspaces.com --repo2-s3-key=<redacted> --repo2-s3-key-secret=<redacted> --repo2-s3-region=nyc3 --repo1-type=posix --repo2-type=s3 --stanza=dev-oltp
2022-08-22 04:03:30.591 P00   INFO: repo1: expire diff backup 20220801-040009F_20220819-040007D
2022-08-22 04:03:30.717 P00   INFO: repo1: remove expired backup 20220801-040009F_20220819-040007D
2022-08-22 04:03:30.756 P00   INFO: repo1: time-based archive retention not met - archive logs will not be expired
2022-08-22 04:03:31.553 P00   INFO: repo2: time-based archive retention not met - archive logs will not be expired
2022-08-22 04:03:31.653 P00 DETAIL: statistics: {"http.client":{"total":2},"http.request":{"total":11},"http.session":{"total":2},"socket.client":{"total":2},"socket.session":{"total":2},"tls.client":{"total":2},"tls.session":{"total":2}}
2022-08-22 04:03:31.653 P00   INFO: expire command end: completed successfully (1066ms)
-------------------PROCESS START-------------------
2022-08-23 04:02:16.965 P00   INFO: expire command begin 2.39: --exec-id=801-b3438787 --log-level-file=detail --repo1-cipher-pass=<redacted> --repo2-cipher-pass=<redacted> --repo1-cipher-type=aes-256-cbc --repo2-cipher-type=aes-256-cbc --repo1-path=/mnt/backup/pgbackrest --repo2-path=/dev/backup/oltp/pgbackrest --repo1-retention-diff=3 --repo2-retention-diff=3 --repo1-retention-full=14 --repo2-retention-full=365 --repo1-retention-full-type=time --repo2-retention-full-type=time --repo2-s3-bucket=tpm-backup --repo2-s3-endpoint=nyc3.digitaloceanspaces.com --repo2-s3-key=<redacted> --repo2-s3-key-secret=<redacted> --repo2-s3-region=nyc3 --repo1-type=posix --repo2-type=s3 --stanza=dev-oltp
2022-08-23 04:02:16.972 P00   INFO: repo1: expire diff backup 20220801-040009F_20220820-040002D
2022-08-23 04:02:17.059 P00   INFO: repo1: remove expired backup 20220801-040009F_20220820-040002D
2022-08-23 04:02:17.122 P00   INFO: repo1: time-based archive retention not met - archive logs will not be expired
2022-08-23 04:02:17.661 P00   INFO: repo2: expire time-based backup 20220801-125242F
2022-08-23 04:02:18.147 P00   INFO: repo2: remove expired backup 20220801-125242F
2022-08-23 04:34:08.314 P00 DETAIL: repo2: 10-1 archive retention on backup 20220809-010304F, start = 0000000600005D3E00000099
2022-08-23 04:34:10.042 P00   INFO: repo2: 10-1 remove archive, start = 0000000600005D3E00000057, stop = 0000000600005D3E00000098
2022-08-23 04:34:10.087 P00   WARN: remote-0 process on 'oltp02' terminated unexpectedly [42]: ERROR: [042]: timeout after 1830000ms waiting for read from 'remote-0'
2022-08-23 04:34:10.088 P00 DETAIL: statistics: {"http.client":{"total":2},"http.request":{"total":384},"http.retry":{"total":1},"http.session":{"total":4},"socket.client":{"total":2},"socket.session":{"total":4},"tls.client":{"total":2},"tls.session":{"total":4}}
2022-08-23 04:34:10.088 P00   INFO: expire command end: completed successfully (1913123ms)
-------------------PROCESS START-------------------
2022-08-24 04:02:12.515 P00   INFO: expire command begin 2.39: --exec-id=18536-a249387f --log-level-file=detail --repo1-cipher-pass=<redacted> --repo2-cipher-pass=<redacted> --repo1-cipher-type=aes-256-cbc --repo2-cipher-type=aes-256-cbc --repo1-path=/mnt/backup/pgbackrest --repo2-path=/dev/backup/oltp/pgbackrest --repo1-retention-diff=3 --repo2-retention-diff=3 --repo1-retention-full=14 --repo2-retention-full=365 --repo1-retention-full-type=time --repo2-retention-full-type=time --repo2-s3-bucket=tpm-backup --repo2-s3-endpoint=nyc3.digitaloceanspaces.com --repo2-s3-key=<redacted> --repo2-s3-key-secret=<redacted> --repo2-s3-region=nyc3 --repo1-type=posix --repo2-type=s3 --stanza=dev-oltp
2022-08-24 04:02:12.520 P00   INFO: repo1: expire diff backup 20220801-040009F_20220821-040009D
2022-08-24 04:02:12.549 P00   INFO: repo1: remove expired backup 20220801-040009F_20220821-040009D
2022-08-24 04:02:12.613 P00   INFO: repo1: time-based archive retention not met - archive logs will not be expired
2022-08-24 04:02:13.128 P00   INFO: repo2: expire time-based backup 20220809-010304F
2022-08-24 04:02:13.571 P00   INFO: repo2: remove expired backup 20220809-010304F
2022-08-24 04:36:25.884 P00 DETAIL: repo2: 10-1 archive retention on backup 20220809-150721F, start = 0000000600005D3E0000009F
2022-08-24 04:36:26.174 P00   INFO: repo2: 10-1 remove archive, start = 0000000600005D3E00000099, stop = 0000000600005D3E0000009E
2022-08-24 04:36:26.223 P00   WARN: remote-0 process on 'oltp02' terminated unexpectedly [42]: ERROR: [042]: timeout after 1830000ms waiting for read from 'remote-0'
2022-08-24 04:36:26.223 P00 DETAIL: statistics: {"http.client":{"total":2},"http.request":{"total":521},"http.retry":{"total":13},"http.session":{"total":6},"socket.client":{"total":2},"socket.session":{"total":6},"tls.client":{"total":2},"tls.session":{"total":6}}
2022-08-24 04:36:26.223 P00   INFO: expire command end: completed successfully (2053708ms)
-------------------PROCESS START-------------------
2022-08-25 04:02:09.196 P00   INFO: expire command begin 2.39: --exec-id=4000-08bdcbf4 --log-level-file=detail --repo1-cipher-pass=<redacted> --repo2-cipher-pass=<redacted> --repo1-cipher-type=aes-256-cbc --repo2-cipher-type=aes-256-cbc --repo1-path=/mnt/backup/pgbackrest --repo2-path=/dev/backup/oltp/pgbackrest --repo1-retention-diff=3 --repo2-retention-diff=3 --repo1-retention-full=14 --repo2-retention-full=365 --repo1-retention-full-type=time --repo2-retention-full-type=time --repo2-s3-bucket=tpm-backup --repo2-s3-endpoint=nyc3.digitaloceanspaces.com --repo2-s3-key=<redacted> --repo2-s3-key-secret=<redacted> --repo2-s3-region=nyc3 --repo1-type=posix --repo2-type=s3 --stanza=dev-oltp
2022-08-25 04:02:09.203 P00   INFO: repo1: expire diff backup 20220801-040009F_20220822-040004D
2022-08-25 04:02:09.231 P00   INFO: repo1: remove expired backup 20220801-040009F_20220822-040004D
2022-08-25 04:02:09.275 P00   INFO: repo1: time-based archive retention not met - archive logs will not be expired
2022-08-25 04:02:10.009 P00   INFO: repo2: time-based archive retention not met - archive logs will not be expired
2022-08-25 04:02:10.109 P00 DETAIL: statistics: {"http.client":{"total":2},"http.request":{"total":9},"http.session":{"total":2},"socket.client":{"total":2},"socket.session":{"total":2},"tls.client":{"total":2},"tls.session":{"total":2}}
2022-08-25 04:02:10.109 P00   INFO: expire command end: completed successfully (914ms)
```

**Comment by Tostino** [comment]
Just a note, at the time this log was taken things were configured to keep 3 diff backups for both repo1 and repo2. Don't think that's the cause, but just wanted to mention that discrepancy.

**Comment by dwsteele** [comment]
Can you reproduce this behavior at debug level so we can see what is going on here? I agree that the repo2 expiration does not look right.

**Comment by dwsteele** [comment]
Never mind, I think I found it.

**Comment by dwsteele** [comment]
I believe this is fixed by PR #1854. Please have a look.

**Comment by Tostino** [comment]
That looks like it'd be the issue. Will upgrade my test env as soon as 2.41 is released
